services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/RouteOptionsTest.java

```diff
@@ -768,6 +769,45 @@ public void emptyExcludeObjectsCleansUpExcludes() {
     assertNull(routeOptions.exclude());
   }
 
+  @Test
+  public void etcPaymentMethod() {
+    RouteOptions routeOptions = routeOptions().toBuilder()
+      .paymentMethodsList(Arrays.asList(DirectionsCriteria.PAYMENT_METHOD_ETC))
+      .build();
+    String query = routeOptions.toUrl("test").getQuery();
+
+    assertEquals("etc", routeOptions.paymentMethods());
+    assertContains(query, "payment_methods=etc");
+  }
+
+  @Test
+  public void generalAndEtcPaymentsMethod() {
+    List<String> paymentMethods = Arrays.asList(
+      DirectionsCriteria.PAYMENT_METHOD_ETC,
+      DirectionsCriteria.PAYMENT_METHOD_GENERAL
+    );
+    RouteOptions routeOptions = routeOptions().toBuilder()
+      .paymentMethodsList(paymentMethods)
+      .build();
+    String query = routeOptions.toUrl("test").getQuery();
+
+    assertEquals("etc,general", routeOptions.paymentMethods());
+    assertEquals(paymentMethods, routeOptions.paymentMethodsList());
+    assertContains(query, "payment_methods=etc%2Cgeneral");
+  }
+
+  @Test
+  public void emptyPaymentMethodsList() {
+    RouteOptions routeOptions = routeOptions().toBuilder()
+      .paymentMethodsList(Collections.<String>emptyList())
```

Comment:
Nope. I just felt confidence in case of null, because it's a default value. I didn't felt confidence in case of an empty list and added the test 🙂 
If you think it's important, I will add a test for the null case 👍

Reply:
What will happen? Will there be a parameter with an empty string or no parameter at all? Since it's no parameter at all for an empty list I guess it will be the same for null so here I don't think it's necessary. I usually add tests for null and empty as a rule, just to be sure. No pressure tho. :)